### ISSUE TITLE
Using win32 to show the splash screen

### ISSUE THREAD

**Issue by lindexi** [dotnet/wpf#2031]
The advantage is that

- Fast
   The sample https://github.com/kkwpsv/SplashImage cost 100ms and the native WPF splash screen cost 300ms
- Can use any file path
   The WPF can only use the resource file
- Fix some issuses
   #297
   #298

**Comment by weltkante** [comment]
WPF is [already using](https://source.dot.net/#WindowsBase/System/Windows/SplashScreen.cs,129) plain win32 to show the splash screen (i.e. the splash screen is not a WPF window)

**Comment by lindexi** [comment]
@weltkante Thank you. But not at all, see https://github.com/kkwpsv/SplashImage and I will edit my summary.

**Comment by weltkante** [comment]
> But not at all

I don't see the difference, both the WPF and your linked example use CreateWindow API to create a native window. Like I said the performance difference must be coming from somewhere else than creating and showing the window.

In fact, if you look at the WPF source I linked and scroll up the resource lookup says it can be pretty slow (200-300ms) so I suspect the fact that WPF wants to localize the splash screen is what makes up for the performance difference.

Maybe adding support for other sources of splash screen images (like you requested) will fix the performance difference.

In either case your issue description is not accurate, WPF already uses win32 API to show the splash screen, what you are asking about is to make it faster and more flexible ...

**Comment by lindexi** [comment]
> In either case your issue title is not accurate, WPF already uses win32 API to show the splash screen, what you are asking about is to make it faster and more flexible

@weltkante Yes, I updated my summary but I do not find a good title.

**Comment by kkwpsv** [comment]
> > But not at all
> 
> I don't see the difference, both the WPF code and your linked example use CreateWindow API to create a native window. They then use different native APIs for showing the image, but that shouldn't make much of a difference, if anything the WPF code using plain GDI should be faster than your sample which first needs to load the GDI+ library. The performance difference must be coming from somewhere else than creating and showing the window.
> 
> In fact, if you look at the WPF source I linked and scroll up the resource lookup [says it can be pretty slow (200-300ms)](https://source.dot.net/#WindowsBase/System/Windows/SplashScreen.cs,110) so I suspect the fact that WPF wants to localize the splash screen is what makes up for the performance difference.
> 
> Maybe adding support for other sources of splash screen images (like you requested) will fix the performance difference. I don't think the window logic itself needs to be changed, just the image lookup.
> 
> In either case your issue title is not accurate, WPF already uses win32 API to show the splash screen, what you are asking about is to make it faster and more flexible ...

@weltkante @lindexi GDI+ is used for gif transparent image. And from [the code](https://source.dot.net/#WindowsBase/System/Windows/SplashScreen.cs,52309ad1ece85c5f) i can see that WPF use WIC to load Image. May this cause performance difference?

**Comment by weltkante** [comment]
Yes loading WIC costs time, but its a native component (no JIT) so probably comparable to loading GDI+ in the other example. Considering the comment I linked already says resource loading slows down by 200-300ms, and OP is saying WPF is ~200ms slower than the other example, there is very little ambiguity where it comes from (assuming the comment isn't outdated). Any further speculation doesn't make much sense without doing actual profiling or measurements.

**Comment by kkwpsv** [comment]
Yes, maybe we should do profiling to get the real reason. 

But now splash image a noticeable delay before it is displayed. While in the example we can see the splash image immediately. **We have a bad experience with delay, and want to see it immediately.** This is the main purpose for this issuses.
